manifests/master/simp_auth.pp

```diff
@@ -1,55 +1,48 @@
 # Add SIMP-specific entries to PuppetServer's auth.conf
 #
+# For documentation about *_allow and *_deny, see the puppetserver docs
+# @see https://puppet.com/docs/puppetserver/5.2/config_file_auth.html#allow-allow-unauthenticated-and-deny
+#
+# @param server_distribution Puppet open source or PE
+#
 # @param auth_conf_path
-#   Type:    Stdlib::AbsolutePath
-#   Default: /etc/puppetlabs/puppetserver/conf.d/auth.conf
 #   The location to the puppet master's auth.conf
 #
 # @param legacy_cacerts_all
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to the PKI cacerts from the legacy `pki` module from all hosts
 #
-# @param legacy_mcollective_all
-#   Type:    Boolean
-#   Default: true
-#   If enabled, allow access to the mcollective cacerts from the legacy `pki` module from all hosts
-#
 # @param legacy_pki_keytabs_from_host
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to each host's own kerberos keytabs from the legacy location
 #
 # @param pki_cacerts_all
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to the cacerts from the `pki_files` module from all hosts
-#
-# @param pki_mcollective_all
-#   Type:    Boolean
-#   Default: true
-#   If enabled, allow access to the mcollective PKI from the `pki_files` module from all hosts
+# @param pki_cacerts_all_allow
+# @param pki_cacerts_all_deny
 #
 # @param keydist_from_host
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to each host's own certs from the `pki_files` module
+# @param keydist_from_host_allow
+# @param keydist_from_host_deny
 #
 # @param krb5_keytabs_from_host
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to each host's own kerberos keytabs from the `pki_files` module
+# @param krb5_keytabs_from_host_allow
+# @param krb5_keytabs_from_host_deny
 #
 class pupmod::master::simp_auth (
   Simplib::ServerDistribution $server_distribution          = simplib::lookup('simp_options::puppet::server_distribution', { 'default_value' => 'PC1' } ),
   Stdlib::AbsolutePath        $auth_conf_path               = '/etc/puppetlabs/puppetserver/conf.d/auth.conf',
   Boolean                     $legacy_cacerts_all           = false,
-  Boolean                     $legacy_mcollective_all       = false,
   Boolean                     $legacy_pki_keytabs_from_host = false,
   Boolean                     $pki_cacerts_all              = true,
-  Boolean                     $pki_mcollective_all          = true,
+  NotUndef                    $pki_cacerts_all_allow        = 'certname',
+  Any                         $pki_cacerts_all_deny         = undef,
   Boolean                     $keydist_from_host            = true,
+  NotUndef                    $keydist_from_host_allow      = '$2',
```

Comment:
$2 is tied to a regex that is not exposed.  Doesn't make sense to me.

Reply:
It's some puppetserver magic. It was also the previous value of the setting, I just moved it.

Reply:
My understanding is that it was the result of the regex which is not exposed.  If so, it doesn't make sense for this to be exposed but the regex to not be exposed. Am I misunderstanding?

Reply:
@jeefberkey Probably should make this `Optional` and `undef` by default so that the user can set it if they want to and default it to `$2` down below to alleviate confusion.

The `@param` documentation needs to be updated so that it makes sense to the user as well.

Reply:
If you are not going to allow the regex line to be configured, why expose the the parameter that goes with it?

Reply:
I'm not opposed to exposing the regex as well.

@trevor-vaughan I'm not sure what benefit that would have

Reply:
@lnemsick-simp So that you can add a different set of allow/deny hosts